base_location/tests/test_completion.py

```diff
@@ -30,20 +40,57 @@ def test_onchange_company_city_completion(self):
         self.assertEqual(self.company.state_id, self.better_zip1.state_id)
         self.assertEqual(self.company.country_id, self.better_zip1.country_id)
 
+    def test_onchange_company_city_id_completion(self):
+        self.company.better_zip_id = self.better_zip2
+        self.company.on_change_city()
+        self.assertEqual(self.company.zip, self.better_zip2.name)
+        self.assertEqual(self.company.city, self.better_zip2.city)
+        self.assertEqual(self.company.state_id, self.better_zip2.state_id)
+        self.assertEqual(self.company.country_id, self.better_zip2.country_id)
+
     def setUp(self):
         super(TestCompletion, self).setUp()
         state_vd = self.env['res.country.state'].create({
             'name': 'Vaud',
             'code': 'VD',
             'country_id': self.ref('base.ch'),
         })
+        self.env['res.country'].browse(self.ref('base.es')).write({
+            'enforce_cities': True
+        })
         self.company = self.env.ref('base.main_company')
         self.better_zip1 = self.env['res.better.zip'].create({
             'name': 1000,
             'city': 'Lausanne',
             'state_id': state_vd.id,
-            'country_id': self.ref('base.ch'),
+            'country_id': self.ref('base.ch')
         })
         self.partner1 = self.env['res.partner'].create({
-            'name': 'Camptocamp',
+            'name': 'Camptocamp'
```

Comment:
Please end all lines that are elements of a dictionary with the trailing comma as a good habit (you have added a lot with it).